pkg/scheduler/framework/plugins/clusteraffinity/types.go

```diff
@@ -6,101 +6,298 @@ Licensed under the MIT license.
 package clusteraffinity
 
 import (
+	"fmt"
+	"math"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 
 	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
 	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
 )
 
-// affinityTerm is a processed version of ClusterSelectorTerm.
-type affinityTerm struct {
-	selector labels.Selector
-}
+const (
+	// resourcePropertyNamePrefix is the prefix (also known as the subdomain) of the label name
+	// associated with all resource properties.
+	resourcePropertyNamePrefix = "resources.kubernetes-fleet.io/"
+
+	// Below are a list of supported capacity types.
+	totalCapacityName       = "total"
+	allocatableCapacityName = "allocatable"
+	availableCapacityName   = "available"
+)
+
+// clusterRequirement is a type alias for ClusterSelectorTerm in the API, which allows
+// easy method extension.
+type clusterRequirement placementv1beta1.ClusterSelectorTerm
 
-// Matches returns true if the cluster matches the label selector.
-func (at *affinityTerm) Matches(cluster *clusterv1beta1.MemberCluster) bool {
-	return at.selector.Matches(labels.Set(cluster.Labels))
+// retrieveResourcePropertyValueFrom retrieves a resource property value from a member cluster.
+//
+// Note that it will return nil if the property is not available for the cluster;
+// the zero value of resource.Quantity, i.e., resource.Quantity{}, is a valid
+// quantity.
+func retrieveResourcePropertyValueFrom(cluster *clusterv1beta1.MemberCluster, name string) (*resource.Quantity, error) {
```

Comment:
nit: a better name? somethign like  retrieveResourceUsage? Not to be similar to "retrievePropertyValueFrom"?

Reply:
Done.